sklearn/linear_model/cd_fast.pyx

```diff
@@ -613,12 +613,12 @@ def enet_coordinate_descent_gram(floating[::1] w,
                     # return if we reached desired tolerance
                     break
 
-
-        with gil:
-            warnings.warn("Objective did not converge."
-            " You might want to increase the number of iterations."
-            " Duality gap: {}, tolerance: {}".format(gap, tol),
-            ConvergenceWarning)
+        else:
```

Comment:
can you add a comment "iterations finished without reaching break" or something like this?
the for-else is a bit unfamiliar to many.

Reply:
yeah I had to look it up myself.

Reply:
I had to double check whether it's triggered by "break" or by "no break" ;)

Reply:
My rule of thumb for things like this is: I assume someone will have to read the code that knows less about scikit-learn and Python than me (like the version of me that joined the project 7 years ago). If I have to look it up, they'll have to look it up (and it will take them longer).

So if you had to look it up, definitely explain what you're doing.